dynamometer-infra/src/main/java/com/linkedin/dynamometer/SimulatedMultiStorageFSDataset.java

```
@@ -0,0 +1,1407 @@
+      SimulatedStorage storage) {
+    BlockListAsLongs.Builder report = BlockListAsLongs.builder();
+    try {
+      for (BInfo b : storage.getBlockMap(bpid).values()) {
+        if (b.isFinalized()) {
+          report.add(b);
+        }
+      }
+    } catch (IOException ioe) {
+      // Ignore
+    }
+    return report.build();
+  }
+
+  @Override
+  public synchronized Map<DatanodeStorage, BlockListAsLongs> getBlockReports(
+      String bpid) {
+    Map<DatanodeStorage, BlockListAsLongs> blockReports = new HashMap<>();
+    for (SimulatedStorage storage : storages) {
+      blockReports.put(storage.getDnStorage(), getBlockReport(bpid, storage));
+    }
+    return blockReports;
+  }
+
+  @Override // FsDatasetSpi
+  public List<Long> getCacheReport(String bpid) {
+    return new LinkedList<Long>();
+  }
+
+  @Override // FSDatasetMBean
+  public long getCapacity() {
+    long total = 0;
+    for (SimulatedStorage storage : storages) {
+      total += storage.getCapacity();
+    }
+    return total;
+  }
+
+  @Override // FSDatasetMBean
+  public long getDfsUsed() {
+    long total = 0;
+    for (SimulatedStorage storage : storages) {
+      total += storage.getUsed();
+    }
+    return total;
+  }
+
+  @Override // FSDatasetMBean
+  public long getBlockPoolUsed(String bpid) throws IOException {
+    long total = 0;
+    for (SimulatedStorage storage : storages) {
+      total += storage.getBlockPoolUsed(bpid);
+    }
+    return total;
+  }
+
+  @Override // FSDatasetMBean
+  public long getRemaining() {
+
+    long total = 0;
+    for (SimulatedStorage storage : storages) {
+      total += storage.getFree();
+    }
+    return total;
+  }
+
+  @Override // FSDatasetMBean
+  public int getNumFailedVolumes() {
+
+    int total = 0;
+    for (SimulatedStorage storage : storages) {
+      total += storage.getNumFailedVolumes();
+    }
+    return total;
+  }
+
+  @Override // FSDatasetMBean
+  public String[] getFailedStorageLocations() {
+    return null;
```

to avoid any potential NPE, maybe better to throw unsupported operation exception, or return an empty array?